integration-tests/framework/v1alpha1/framework.go

```diff
@@ -0,0 +1,187 @@
+	"github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+	pod "github.com/openebs/maya/pkg/kubernetes/pod/v1alpha1"
+
+	validatehook "github.com/openebs/maya/pkg/kubernetes/webhook/validate/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	// auth plugins
+	//_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	// DefaultTimeOut is the default time in seconds
+	// for Eventually block
+	DefaultTimeOut int = 600
+	// DefaultPollingInterval is the default polling
+	// time in seconds for the Eventually block
+	DefaultPollingInterval int = 10
+	// NodeCountOne is the 1 node
+	// needed to run this test
+	NodeCountOne int = 1
+)
+
+var (
+	defaultOpenebsComponents []*installer.DefaultInstaller
+)
+
+// Framework supports common operations used by tests
+type Framework struct {
+	BaseName string
+	// configuration for framework's options
+	Options FrameworkOptions
+}
+
+// FrameworkOptions provide different options specfic to tests
+type FrameworkOptions struct {
+	MinNodeCount int
+	Artifacts    artifacts.ArtifactSource
+}
+
+//func NewFrameworkDefault(baseName string, MinNodeCount int) *Framework {
+//	options := FrameworkOptions{
+//		MinNodeCount: MinNodeCount,
+//	}
+//	return NewFramework(baseName, options)
+//}
+
+// NewFrameworkDefault makes a framework using default values and sets
+// up a BeforeEach/AfterEach
+func NewFrameworkDefault(baseName string, options ...FrameworkOptions) *Framework {
+	Options := FrameworkOptions{}
+	for _, value := range options {
+		Options = FrameworkOptions{
+			MinNodeCount: value.MinNodeCount,
+			Artifacts:    value.Artifacts,
+		}
+	}
+	return NewFramework(baseName, Options)
+}
+
+// NewFramework creates a test framework.
+func NewFramework(baseName string, options FrameworkOptions) *Framework {
+	f := &Framework{
+		BaseName: baseName,
+		Options:  options,
+	}
+
+	BeforeEach(f.BeforeSuite)
+	AfterEach(f.AfterSuite)
+
+	return f
+}
+
+// BeforeSuite installs openebs control plane
+// components
+func (f *Framework) BeforeSuite() {
+	// Fetching the kube config path
+	configPath, err := kubernetes.GetConfigPath()
+	Expect(err).ShouldNot(HaveOccurred())
+
+	// Setting the path in environemnt variable
+	err = os.Setenv(string(v1alpha1.KubeConfigEnvironmentKey), configPath)
+	Expect(err).ShouldNot(HaveOccurred())
+	// Getting clientset
+	cl, err := kubernetes.GetClientSet()
+	Expect(err).ShouldNot(HaveOccurred())
+
+	// Checking appropriate node numbers
+	nodes, _ := cl.CoreV1().Nodes().List(v1.ListOptions{})
+	Expect(nodes.Items).Should(HaveLen(f.Options.MinNodeCount))
+
+	// Fetching the openebs component artifacts
+	artifactsOpenEBS, errs := artifacts.GetArtifactsListUnstructuredFromFile(f.Options.Artifacts)
+	Expect(errs).Should(HaveLen(0))
+
+	By("Deploying OpenEBS components in openebs namespace")
+	for _, artifact := range artifactsOpenEBS {
+		openebsBuilder := installer.BuilderForObject(artifact)
```

Comment:
In next frame work PR this installation should be implemented in builder pattern.
Mostly like
```
installerObj := NewTestInstaller().
		           WithArtifact(artifact).
			    GetUnstructObj().
			     GetInstallerObj().
			      Install()
```
cc: @AmitKumarDas @prateekpandey14